src/main/java/randoop/resource/ClassAnnotationScanner.java

```diff
@@ -0,0 +1,148 @@
+package randoop.resource;
+
+import java.util.Collection;
+import java.util.HashSet;
+import java.util.Set;
+import org.objectweb.asm.AnnotationVisitor;
+import org.objectweb.asm.ClassVisitor;
+import org.objectweb.asm.MethodVisitor;
+import org.objectweb.asm.Opcodes;
+import org.objectweb.asm.TypePath;
+import org.plumelib.reflection.Signatures;
+
+/**
+ * ClassAnnotationScanner is a ClassVisitor that collects methods that are annotated with specific
+ * annotations.
+ */
+public class ClassAnnotationScanner extends ClassVisitor {
+  /** OpCode for ASM used throughout this class. */
+  private static final int ASM_OPCODE = Opcodes.ASM7;
+
+  /** The desired annotations that serve as the criteria for methods we want to capture. */
+  private final Collection<String> desiredAnnotations;
+
+  /** The Randoop fully qualified class name of the class being visited. */
+  private String fullyQualifiedClassName = null;
+
+  /** The accumulated set of captured fully qualified method signatures. */
+  private final Set<String> matchingFullyQualifiedMethodSignatures = new HashSet<>();
```

Comment:
Why is this a set rather than a list?  Is there a possibility of duplicates?  (Maybe if a method contains both `@SideEffectFree` and `@Pure`?)

Reply:
Yes, that is the intent for that very reason (which I'm not sure actually arises, but to my knowledge I didn't see any guarantees that `@Pure` and `@SideEffectFree` were mutually exclusive.